### PR TITLE
CANS-771. Print comments at the item level

### PR DESCRIPTION
**Pull request by denys-davydov**
## JIRA
https://osi-cwds.atlassian.net/browse/CANS-771

## Description
1. Fix a bug with losing the i18n data while printing
2. Add printing item comments with an assessment printing

## Tests

- [x] I used TDD to develop the feature
- [x] I have included unit tests
- [ ] I have included new acceptance tests or modified existing ones
- [ ] I have included other tests

## Types of changes

- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (No behavioral changes)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] I ran all my tests locally which were green.
- [x] My code adds no new issues to Code Climate
- [x] I ran all the linters for the project.
- [x] My code is in a stable state ready to be deployable, but not necessarily complete.
- [x] I promise on my honor as a CWDS developer to ensure I don't break the build (or bring donuts if I leave things broken), to check linters, use best practices, to leave the code in better shape than I found it, and to have a good day.

### CODE REVIEW

File: app/javascript/Application/components/Assessment/AssessmentContainer.js [hunk 1]
```diff
@@ -50,13 +50,8 @@ class AssessmentContainer extends Component {
...
 
   async componentDidMount() {
-    this.initHeaderButtons(this.state.isSaveButtonEnabled)
     const assessmentId = this.props.match.params.id
-    let isEditable = assessmentId === undefined
-    if (!isEditable) {
-      isEditable = await SecurityService.checkPermission(`assessment:write:${assessmentId}`)
-    }
-    this.setState({ isEditable: isEditable })
+    await this.updateIsEditableState(assessmentId)
```
Comment (denys-davydov) [inline app/javascript/Application/components/Assessment/AssessmentContainer.js]: This is a small refactoring

File: app/javascript/Application/components/Assessment/AssessmentContainer.js [hunk 2]
```diff
@@ -168,10 +169,11 @@ class AssessmentContainer extends Component {
     }
   }
 
-  onFetchI18nSuccess(i18n) {
-    this.setState({
+  async onFetchI18nSuccess(i18n) {
+    await this.setState({
       i18n: i18n,
     })
+    this.initHeaderButtons(this.state.isSaveButtonEnabled)
```
Comment (denys-davydov) [inline app/javascript/Application/components/Assessment/AssessmentContainer.js]: So the header buttons are updated only after i18n is fetched, it fixes the bug with losing i18n while printing

**Comment by nasar-cambria** [comment]
@denys-davydov If I try to print when the form is in progress it gives me console errors.
```
TypeError: Cannot read property 'name' of undefined
```

**Comment by denys-davydov** [comment]
@wmitchell-cambria Here is an extract from coverage result:
```
PrintAssessment.js                                                  |    95.35 |    82.61 |      100 |      100 |... 13,120,165,204 |
```

**Comment by denys-davydov** [comment]
@nasar-cambria Could not reproduce the issue

**Comment by denys-davydov** [comment]
@nasar-cambria The issue you've raised was confirmed and a bug ticket was created for it https://osi-cwds.atlassian.net/browse/CANS-835

**Pull request merged**